site/source/docs/porting/Debugging.rst

```diff
@@ -137,6 +137,58 @@ Debug printouts can even execute arbitrary JavaScript. For example::
   }
 
 
+Printing exception messages
+===========================
+
+Exceptions are thrown from WebAssembly using exception pointers, which means
+that try/catch/finally blocks in JavaScript will only receive a number, which
+represents a pointer into linear memory. In order to get the exception message,
+the user will need to create some WASM code which will extract the meaning from
+the exception. In the example code below we created a function that receives an
+``int`` which is pointer to an ``std::exception``, and by casting the pointer
+returns the ``what`` function call result.
+
+.. code-block:: cpp
+
+  #include <bind.h>
+
+  std::string getExceptionMessage(int exceptionPtr) {
+    return std::string(reinterpret_cast<std::exception *>(exceptionPtr)->what());
+  }
+
+  EMSCRIPTEN_BINDINGS(Bindings) {
+    emscripten::function("getExceptionMessage", &getExceptionMessage);
+  };
+
+Once such a function has been created, exception handling code in javascript
+can call it when receiving an exception from WASM. Here the function is used
+in order to log the thrown exception.
+
+.. code-block:: javascript
+
+  try {
+    ... // some code that calls WebAssembly
+  } catch (exception) {
+    console.error(Module.getExceptionMessage(exception));
+  } finally {
+    ...
+  }
+
+It's important to notice that this example code will work only for thrown
+statically allocated exceptions. If your code throws other objects, such as
+strings or dynamically allocated exceptions, the handling code will need to
+take that into account. For example, if your code throws either exception
+pointers or strings, the javascript code can handle these situations like
+this:
+
+.. code-block:: javascript
+
+  function getExceptionMessage(exception: any): any {
+    return typeof exception === 'number'
```

Comment:
But since the exception is a pointer won't it always appear here as a number (i32) , regardless of what one throws?  What other type could this be here?

Reply:
That depends on the code you wrap. I think that if you use embind you might also receive JS exceptions, and also - if your try clause wraps non-WASM code, you'll need some way to differentiate the errors.

AFAIK, actual exceptions from WASM are always number.